File: 0905-length-of-longest-fibonacci-subsequence/0905-length-of-longest-fibonacci-subsequence.cpp
class Solution {
public:
    int lenLongestFibSubseq(vector<int>& arr) {
        set<int>st;
        for(auto i: arr)st.insert(i);
        int n=arr.size();
        int maxcount=0;
        for(int i=0;i<n;i++){
            for(int j=i+1;j<n;j++){
                int a=arr[i];
                int b=arr[j];
                int count=2;
                while(st.find(a+b)!=st.end()){
                    int c=a+b;
                    a=b;
                    b=c;
                    count++;
                }
                maxcount=max(maxcount,count);
            }
        }
        return (maxcount<3)?0:maxcount;
    }
};

File: 0905-length-of-longest-fibonacci-subsequence/0905-length-of-longest-fibonacci-subsequence.py
class Solution(object):
    def lenLongestFibSubseq(self, arr):
        """
        :type arr: List[int]
        :rtype: int
        """
        s=set()
        for i in arr:
            s.add(i)
        p=len(arr)
        ans=0
        for i in range(0,p,1):
            for j in range(i+1,p,1):
                a=arr[i]
                b=arr[j]
                count=2
                while (a+b) in s:
                    c=a+b
                    a=b
                    b=c
                    count=count+1
                ans=max(ans,count)
        
        if ans>2: return ans
        return 0
        